sky/data/skystorage.py

```diff
@@ -0,0 +1,227 @@
+"""Skystorage module"""
+import click
+import contextlib
+import filelock
+import os
+import psutil
+import subprocess
+import time
+from typing import Dict, Tuple
+
+from sky import sky_logging
+
+logger = sky_logging.init_logger(__name__)
+
+C_SYNC_FILE_PATH = '~/.skystorage'
+
+
+@click.group()
+def main():
+    pass
+
+
+def update_interval(interval: int, elapsed_time: int):
+    diff = interval - elapsed_time
+    if diff <= 0:
+        return 0
+    else:
+        return diff
+
+
+def is_locked(file_path: str):
+    """Checks if the given lock file is locked"""
+    locked = False
+    if os.path.exists(file_path):
+        lock = filelock.FileLock(file_path)
+        try:
+            lock.acquire(timeout=0)
+        except filelock.Timeout:
+            locked = True
+        finally:
+            if not locked:
+                lock.release()
+    return locked
+
+
+def set_s3_sync_cmd(src_path: str, bucketname: str, num_threads: int,
+                    delete: bool, no_follow_symlinks: bool):
+    """Builds sync command for aws s3"""
+    config_cmd = ('aws configure set default.s3.max_concurrent_requests '
```

Comment:
This seems not only applied to the following commands but other s3 related commands as well. For example, it may affect the mount mode and copy mode. Is there a way to limit the change to the current command only?

Reply:
> In COPY mode, this method remains unaffected because _execute_storage_csync is executed after all files required for the COPY mode have been downloaded to the remote node.

This is not true if a user runs `sky launch` on the same cluster twice right? 

Resetting it after every `run_sync` sounds good to me.

Reply:
Currently, Skypilot use only default number of threads(10) for running csync as there are minuscule difference by increasing the number of threads on large files. Especially enabling crt client and s5cmd would make the benefit from adjusting threads unnecessary. For simplicity, I'll remove the portion related to adjusting number of threads including `f'aws configure set default.s3.max_concurrent_requests  {num_threads}'`. 

aws s3 sync with 1 thread on 6 of 10GB files: 2897.0470000505447
aws s3 sync with 10 threads on 6 of 10GB files: 612.9618566036224
aws s3 sync with 20 threads on 6 of 10GB files: 685.7049639225006
aws s3 sync with 10 threads with CRT client on 6 of 10GB files: 412.372611364
s5cmd sync on 6 of 10GB files: 386.55710673332214
s5cmd sync with 10 threads with CRT client on 6 of 10GB files: 108.94380240440368